src/porepy/models/fluid_property_library.py

```diff
@@ -0,0 +1,396 @@
+            interfaces: List of interface grids.
+
+        Returns:
+            Discretization for the interface mobility.
+
+        """
+        return pp.ad.UpwindCouplingAd(self.mobility_keyword, interfaces)
+
+
+class ConstantViscosity(PorePyModel):
+    """Constant viscosity for a single-phase fluid."""
+
+    def viscosity_of_phase(self, phase: pp.Phase) -> ExtendedDomainFunctionType:
+        """Mixin method for :class:`~porepy.compositional.compositional_mixins.
+        FluidMixin` to provide a constant viscosity for the fluid's phase.
+
+        Parameters:
+            phase: The single fluid phase.
+
+        Returns:
+            A function representing representing the constant phase viscosity on some
+            domains.
+
+        """
+
+        def mu(domains: pp.SubdomainsOrBoundaries) -> pp.ad.Operator:
+            return Scalar(self.fluid.reference_component.viscosity, "viscosity")
+
+        return mu
+
+
+class ConstantFluidThermalConductivity(PorePyModel):
+    """Ïmplementation of a constant thermal conductivity for a single-phase fluid."""
+
+    def thermal_conductivity_of_phase(
+        self, phase: pp.Phase
+    ) -> ExtendedDomainFunctionType:
+        """Mixin method for :class:`~porepy.compositional.compositional_mixins.
+        FluidMixin` to provide a constant thermal conductivity for the fluid's phase.
+
+        Parameters:
+            phase: The single fluid phase.
+
+        Returns:
+            A function representing the constant phase conductivity on some domains.
+
+        """
+
+        def kappa(domains: pp.SubdomainsOrBoundaries) -> pp.ad.Operator:
+            return Scalar(
+                self.fluid.reference_component.thermal_conductivity,
+                "fluid_thermal_conductivity",
+            )
+
+        return kappa
+
+    def normal_thermal_conductivity(
+        self, interfaces: list[pp.MortarGrid]
+    ) -> pp.ad.Scalar:
+        """Constant normal thermal conductivity of the fluid given by the fluid
+        constants stored in the fluid's reference component.
+
+        Using the fluid value corresponds to assuming a fluid-filled fracture.
+
+        Parameters:
+            interfaces: List of interface grids.
+
+        Returns:
+            Operator representing normal thermal conductivity on the interfaces.
+
+        """
+        # NOTE this is not really a fluid-related const. law, it is more related to
+        # mixed-dimensional problems.
+        val = self.fluid.reference_component.normal_thermal_conductivity
+        return Scalar(val, "normal_thermal_conductivity")
+
+
+class FluidEnthalpyFromTemperature(PorePyModel):
+    """Implementation of a linearized fluid enthalpy :math:`c(T - T_{ref})` for a
+    single-phase, single-component fluid.
+
+    It uses the specific heat capacity of the fluid's reference component as :math:`c`,
+    which is constant.
+
+    """
+
+    def fluid_specific_heat_capacity(self, subdomains: list[pp.Grid]) -> pp.ad.Operator:
```

Comment:
Would it not make sense to allow boundaries?

Reply:
It would. Should we bother, despite this being called only inside the scope of the enthalpy function?

Reply:
I can see this being useful for setting BCs. I suggest we make an issue that takes note of this, together with other unresolved comments.

Reply:
Created #1259